src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_FlipView.cs

```diff
@@ -121,6 +124,41 @@ public async Task When_Flipview_Items_Modified()
 
 		}
 
+		[TestMethod]
+		public async Task When_Flipview_DataTemplateSelector()
+		{
+			var dataContext = new When_Flipview_DataTemplateSelector_DataContext();
+
+			var page = new FlipView_TemplateSelectorPage();
+			page.DataContext = dataContext;
+
+			WindowHelper.WindowContent = page;
+			await WindowHelper.WaitForLoaded(page);
+
+			FlipView SUT = page.MyFlipView;
+
+			for (var i = 0; i < SUT.Items.Count; i++)
+			{
+				var item = SUT.Items[i];
+				Assert.AreEqual(SUT.SelectedIndex, i, "Has the SelectedIndex the expected value?");
+
+				var textBlockName = Convert.ToInt32(item) % 2 == 0 ? "TextPair" : "TextOdd";
 
+				var textblock = SUT.FindName(textBlockName) as TextBlock;
+				Assert.IsNotNull(textblock, "Was the expected Template was applied?");
+				Assert.AreEqual(item.ToString(), textblock?.Text, "Has the TextBlock the expected value?");
+
+				if (i < SUT.Items.Count-1)
+				{
+					SUT.SelectedIndex = i + 1;
+					await WindowHelper.WaitForIdle();
+				}
```

Comment:
could move this block to the beginning of the `for` block, 
and use `if (SUT.SelectedIndex != i)`

Reply:
By doing so, we would not have the expected results.

```csharp
SUT.SelectedIndex = i + 1;
```

triggers the FlipView to change the SelectedItem

Reply:
I mean like:
```
for (var i = 0; i < SUT.Items.Count; i++)
{
    if (SUT.SelectedIndex != i)
	{
		SUT.SelectedIndex = i;
		await WindowHelper.WaitForIdle();
	}

	// the rest
}
```
this will make the code read more natural, so you arent doing `for {verify x; scroll to x+1}`, but `for {scroll to x; verify x}`

Reply:
@ajpinedam was this comment addressed ?

Reply:
comments were addressed @Xiaoy312